### PR TITLE
Add Dgecon and tests, plus small fix to Dlacn2

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by kortschak** [comment]
That time-out during TestDgetrs is something I've seen before - I don't understand why the cgo tests take so long (the native tests are typically under ~10s for the long running ones here):

```
=== RUN TestDlange-2
--- PASS: TestDlange-2 (0.04 seconds)
=== RUN TestDpotrf-2
--- PASS: TestDpotrf-2 (6.29 seconds)
=== RUN TestDgecon-2
--- PASS: TestDgecon-2 (0.00 seconds)
=== RUN TestDgelq2-2
--- PASS: TestDgelq2-2 (0.00 seconds)
=== RUN TestDgels-2
--- PASS: TestDgels-2 (266.43 seconds)
=== RUN TestDgelqf-2
--- PASS: TestDgelqf-2 (105.36 seconds)
=== RUN TestDgeqr2-2
--- PASS: TestDgeqr2-2 (0.00 seconds)
=== RUN TestDgeqrf-2
--- PASS: TestDgeqrf-2 (87.53 seconds)
=== RUN TestDgetf2-2
--- PASS: TestDgetf2-2 (0.00 seconds)
	dgetf2.go:97: Returned ok with singular matrix.
	dgetf2.go:97: Returned ok with singular matrix.
=== RUN TestDgetrf-2
--- PASS: TestDgetrf-2 (28.46 seconds)
=== RUN TestDgetrs-2
SIGQUIT: quit
PC=0x555923
[snip]
*** Test killed with quit: ran too long (10m0s).
```

File: native/dgecon.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+// Dgecon estimates the reciprocal of the condition number of the n×n matrix A
+// given the LU decomposition of the matrix. The condition number computed may
+// be based on the 1-norm or the ∞-norm.
+//
+// The slice a contains the result of the LU decomposition of A as computed by Dgetrf.
+//
+// anorm is the corresponding 1-norm or ∞-norm of the original matrix A.
+//
+// work is a temporary data slice of length at least 4*n and will panic otherwise.
```
Comment (kortschak) [inline native/dgecon.go]: work will not panic - "and Dgecon will panic otherwise."

Also below.

**Comment by btracey** [comment]
Does that ever happen to you locally? For me the full native suite runs in about 10s and the cgo suite runs in about 4.

**Comment by kortschak** [comment]
Never happens to me. I think it is something to do with Travis hardware and the way they deal with floats, but I have no evidence for that.

This LGTM with minor comments. Much clearer than the fortran which manages to make spaghetti out of 50 line routine.

**Pull request merged**